m4/pdns_enable_reproducible.m4

```diff
@@ -12,7 +12,9 @@ AC_DEFUN([PDNS_ENABLE_REPRODUCIBLE], [
     AC_DEFINE([REPRODUCIBLE], [1], [Define to 1 for reproducible builds])
   ],[
     build_user=m4_esyscmd_s(id -u -n)
-    build_host=m4_esyscmd_s(hostname -f || hostname)
+    AS_IF([test x"$host_os" = "xSunOS"],[
+      build_host=m4_esyscmd_s(uname -srvp)],[
```

Comment:
this returns `SunOS 5.9 Generic_118558-39 sparc` which does not look like a hostname